### DISCUSSION TITLE
Volumes supporting.

### DISCUSSION THREAD

**Discussion by AksutikD** [dasniko/testcontainers-keycloak#80] [Q&A] [answered]
Hi @dasniko, I'm using your project to test keycloak app with customised auth flow.

I successfully added custom spi to test container, but faced with the issue:
How to map correctly custom themes and other files from my resources, I need it because the auth flow contains custom templates and uses custom settings.
I checked test container in terminal, providers folder contains all required jars without .properties files and theme folder is empty.

My project docker-compose contains the following volumes:
volumes:
      - ./target/classes/themes/custom:/opt/keycloak/themes/custom
      - ./target/classes/appsettings.properties:/opt/keycloak/providers/appsettings.properties:z
      - ./target/classes/appsettings-dev.properties:/opt/keycloak/providers/appsettings-dev.properties:z
      - ./target/classes/CA.pem:/opt/keycloak/providers/CA.pem:z
      - ./target/custom-sms-authenticator.1.0.0.jar:/opt/keycloak/providers/custom-sms-authenticator.1.0.0.jar:z

    //Current KeycloakContainer setup 
    private static final KeycloakContainer keycloak = new KeycloakContainer()
            .withProviderLibsFrom(Maven.resolver()
                    .loadPomFromFile("pom.xml")
                    .resolve("org.springframework.boot:spring-boot-devtools",
                            "uk.org.webcompere:lightweight-config",
                            "jakarta.enterprise:jakarta.enterprise.cdi-api",
                            "io.confluent:kafka-json-schema-serializer")
                    .withTransitivity().asList(File.class))
            .withProviderClassesFrom("target/classes")
            .withRealmImportFile("realm-export.json");

I didn't found how to add those volumes. Is there correct way to map volumes to our test container? 
Thanks!

**Comment by dasniko** [comment, marked as answer]
Arbitrary volume mapping is not related to the Keycloak container, but to the core Testcontainers framework. See the official docs on how to map volumes to your container: https://www.testcontainers.org/features/files/

The Keycloak testcontainer only offers some convenient wrapper methods to add custom classes and custom dependencies. These methods also rely on the core mapping features.

**Comment by AksutikD** [comment]
I got it, Niko! Thanks for your time, sorry for carelessness. Found the answer later :)
